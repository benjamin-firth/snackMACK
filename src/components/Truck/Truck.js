import React from 'react';
import { addPotentialLocation } from '../../actions/index'
import { connect } from 'react-redux';
import './Truck.scss';
import imagecomingsoon from '../../styles/imagecomingsoon.png';

export const Truck = ({ name, desc, image, addPotentialLocation }) => {

  const submitPotentialLocation = () => {
    const location = { name, desc, image };
    addPotentialLocation(location);
  }

  return (
    <article className='truck-card'>
      <h4>{name}</h4>
      <hr/>
      {image ? <img src={image.logo} alt='Food truck logo' height='100' width='170' /> : <img src={imagecomingsoon} alt='Food truck logo' height='100' width='170' />}
      <p>{desc}</p>
      <button 
        type='button' 
        onClick={submitPotentialLocation}
        >ADD TO POTENTIAL LOCATIONS</button>
    </article>
  )
}

const mapDispatchToProps = dispatch => ({
  addPotentialLocation: location => dispatch(addPotentialLocation(location))
})

export default connect(null, mapDispatchToProps)(Truck);