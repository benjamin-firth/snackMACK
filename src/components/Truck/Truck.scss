.truck-card {
  background-color: #EEEDF0;
  border: 2px solid #83A1B0;
  border-radius: 10px;
  margin-bottom: 20px;
  padding: 10px;

  hr {
    margin: 3px 0px 8px 0px;
  }

  img {
    margin-bottom: 5px; 
  }
}