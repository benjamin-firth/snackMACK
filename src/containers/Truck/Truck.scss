.truck-card {
  align-items: center;
  background-color: #EEEDF0;
  border: 2px solid #83A1B0;
  border-radius: 10px;
  cursor: pointer;
  display: flex; 
  flex-direction: column;
  margin-bottom: 20px;
  padding: 10px 10px 0px 10px;

  h4 {
    font-size: 20px;
  }

  hr {
    margin: 3px 0px 8px 0px;
  }

  img, p {
    margin-bottom: 5px; 
  }
}