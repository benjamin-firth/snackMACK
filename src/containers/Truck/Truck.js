import React, { Component } from 'react';
import { togglePotentialLocation } from '../../actions/index'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './Truck.scss';
import imagecomingsoon from '../../images/imagecomingsoon.png';

export class Truck extends Component {
  submitPotentialLocation = () => {
    this.props.togglePotentialLocation(this.props.truck)
  }

  render() {
    const { name, desc, image, isPotentialLocation } = this.props.truck;

    return (
      <article className='truck-card'>
        <h4>{name}</h4>
        <hr/>
        {image ? <img src={image.logo} alt='Food truck logo' height='100' width='170' /> : <img src={imagecomingsoon} alt='Food truck logo' height='100' width='170' />}
        <p>{desc}</p>
        <button 
          type='button' 
          onClick={this.submitPotentialLocation}
          >{isPotentialLocation ? 'REMOVE LOCATION' : 'ADD TO POTENTIAL LOCATIONS'}</button> 
      </article>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  togglePotentialLocation: truck => dispatch(togglePotentialLocation(truck))
})

Truck.propTypes = {
  truck: PropTypes.string
}

export default connect(null, mapDispatchToProps)(Truck);