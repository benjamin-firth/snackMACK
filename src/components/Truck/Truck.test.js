import React from 'react';
import { shallow } from 'enzyme';
import { Truck } from './Truck';

describe('Truck', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <Truck truck='test' />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

})