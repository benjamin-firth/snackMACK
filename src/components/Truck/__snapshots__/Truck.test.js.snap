// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Truck should match the snapshot 1`] = `
<article
  className="truck-card"
>
  <h4 />
  <hr />
  <img
    alt="Food truck logo"
    height="100"
    src="imagecomingsoon.png"
    width="170"
  />
  <p />
  <button
    onClick={[Function]}
    type="button"
  >
    ADD TO POTENTIAL LOCATIONS
  </button>
  <button
    onClick={[Function]}
    type="button"
  >
    REMOVE LOCATION
  </button>
</article>
`;
